packages/perspective-viewer-d3fc/src/js/axis/crossAxis.js

```diff
@@ -73,46 +73,68 @@ export const domain = settings => {
     return _domain;
 };
 
-export const labelFunction = settings => {
-    switch (axisType(settings)) {
+export const labelFunction = (settings, labelField = "__ROW_PATH__", typeField = "crossValues") => {

```

Comment:
This is looking better.

My only reservation is that `labelField` and `typeField` don't always have the same meaning. Here, `labelField` is referring to a field in the data array (which is similar in meaning to `valueName` from `mainAxis`), and `typeField` refers to a property of the settings structure (containing a list of field names). In some of the other functions `labelField` is used to refer to the field-name list in the settings structure.

Reply:
I have renamed the fields to valueName (when the field is coming from the data array) and settingName when it is referring to a field name list in the settings structure.